controllers/cloud.redhat.com/clowdenvironment_controller.go

```diff
@@ -65,9 +65,6 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/source"
 )
 
-var mu sync.RWMutex
```

Comment:
The mutex stuff was from 83fc721 "Stop competition: App vs Env". Looks like it was added to ensure that a given app's environment was not being actively reconciled when the app was also being reconciled.  I think we should keep this in.